apps/sms/services/js/moz_mobile_message/messaging_serivce.js

```diff
@@ -0,0 +1,176 @@
+/*global BridgeServiceMixin, Settings */
+'use strict';
+
+if (!('BridgeServiceMixin' in self)) {
+  importScripts('/services/js/bridge_service_mixin.js');
+}
+
+(function(exports) {
+  const SERVICE_NAME = 'messaging-service';
+  const CLIENT_NAME = 'mozMobileMessageShim';
+
+  const METHODS = Object.freeze([
+    'sendSMS', 'sendMMS', 'resendMessage'
+  ]);
+
+  // 0 is a valid value so we need to take care to not consider it as a falsy
+  // value. We want to return null for anything that's not a number or a
+  // string containing a number.
+  function sanitizeServiceId(serviceId) {
+    if (serviceId == null || // null or undefined
+        isNaN(+serviceId)) {
+      serviceId = null;
+    } else {
+      serviceId = +serviceId;
+    }
+
+    return serviceId;
+  }
+
+  function getSendOptionsFromServiceId(serviceId) {
+    var sendOpts;
+
+    if (serviceId != null && // not null, not undefined
+        Settings.hasSeveralSim()) {
+      sendOpts = { serviceId: serviceId };
+    }
+
+    return sendOpts;
+  }
+
+  var MessagingService = {
+    /**
+     * Initializes our service using the Service mixin's initService and
+     * requests mozMobileMessage shim as client.
+     */
+    init() {
+      this.initService();
+
+      if (!('bridge' in self)) {
+        importScripts('/lib/bridge.js');
+      }
+      this.client = bridge.client(CLIENT_NAME);
+    },
+
+    /**
+     * Send sms by using send API shim.
+     * @param {*} opts Options for sending sms contains following properties:
+     * - recipients (string or array of string): contains the list of
+     *   recipients for this message.
+     * - content (string): the message's body.
+     * - serviceId (optional long or string): the SIM serviceId we use to send
+     *   the message.
+     * @returns {Promise} Promise with array of the request result.
+     */
+    sendSMS: function mm_send(opts) {
+      var recipients = opts.recipients || [],
+          content = opts.content,
+          serviceId = sanitizeServiceId(opts.serviceId);
+
+      if (!Array.isArray(recipients)) {
+        recipients = [recipients];
+      }
+
+      var i = 0;
+      var sendOpts = getSendOptionsFromServiceId(serviceId);
+      var requests = [];
+
+      recipients.forEach((recipient) => {
+        requests.push(this.client('send', recipient, content, sendOpts));
```

Comment:
You mean, we should clarify what is returned from `this.client('send', recipient, content, sendOpts)` because it might be simply the DOMRequest, not the result of the DOMRequest?